bpf/pca.h

```diff
@@ -0,0 +1,106 @@
+#ifndef __PCA_H__
+#define __PCA_H__
+
+#include "utils.h"
+#include <string.h>
+
+static int attach_packet_payload(void *data, void *data_end, struct __sk_buff *skb, u64 current_time ){
+        payload_meta meta;
+        u64 flags = BPF_F_CURRENT_CPU;
+        // Enable the flag to add packet header
+        // Packet payload follows immediately after the meta struct
+        u32 packetSize = (__u32)(data_end-data);
+        // For packets which are allocated non-linearly struct __sk_buff does not necessarily 
+        // has all data lined up in memory but instead can be part of scatter gather lists. 
+        // This command pulls data from the buffer but incurs data copying penalty.
+        if (packetSize < skb->len){
+            packetSize = skb->len;
+            if (bpf_skb_pull_data(skb, skb->len)){
+                return TC_ACT_UNSPEC;
```

Comment:
I think we come to this case for TCP DNS have u tried to capture DNS over TCP and measured the CPU usage ? u can simply do `dig google.om +tcp` to do DNS over TCP

Reply:
I am not sure what is happening here, but when I experiment, the requests/response generated via dig google.com `+tcp` are also UDP packets.

Reply:
no that is indeed tcp dns I have used it

Reply:
Then, I am able to capture those.